src/IssueViz.Security/Taint/SonarQubeIssueToIssueVisualizationConverter.cs

```diff
@@ -0,0 +1,127 @@
+        private readonly IAbsoluteFilePathLocator absoluteFilePathLocator;
+
+        [ImportingConstructor]
+        public SonarQubeIssueToIssueVisualizationConverter(IAnalysisIssueVisualizationConverter issueVisualizationConverter, IAbsoluteFilePathLocator absoluteFilePathLocator)
+        {
+            this.issueVisualizationConverter = issueVisualizationConverter;
+            this.absoluteFilePathLocator = absoluteFilePathLocator;
+        }
+
+        public IAnalysisIssueVisualization Convert(SonarQubeIssue sonarQubeIssue)
+        {
+            var analysisIssue = ConvertToAnalysisIssue(sonarQubeIssue);
+            var issueViz = issueVisualizationConverter.Convert(analysisIssue);
+
+            return issueViz;
+        }
+
+        private IAnalysisIssueBase ConvertToAnalysisIssue(SonarQubeIssue sonarQubeIssue)
+        {
+            if (sonarQubeIssue.TextRange == null)
+            {
+                throw new ArgumentNullException(nameof(sonarQubeIssue.TextRange));
+            } 
+
+            var localFilePath = absoluteFilePathLocator.Locate(sonarQubeIssue.FilePath);
+
+            return new AnalysisIssue(
+                sonarQubeIssue.RuleId,
+                Convert(sonarQubeIssue.Severity),
+                AnalysisIssueType.Vulnerability,
+                sonarQubeIssue.Message,
+                localFilePath,
+                sonarQubeIssue.TextRange.StartLine,
+                sonarQubeIssue.TextRange.EndLine,
+                sonarQubeIssue.TextRange.StartOffset,
+                sonarQubeIssue.TextRange.EndOffset,
+                sonarQubeIssue.Hash,
+                Convert(sonarQubeIssue.Flows)
+            );
+        }
+
+        private IReadOnlyList<IAnalysisIssueFlow> Convert(IEnumerable<IssueFlow> flows) =>
+            flows.Select(x => new AnalysisIssueFlow(Convert(x.Locations))).ToArray();
+
+        private IReadOnlyList<IAnalysisIssueLocation> Convert(IEnumerable<IssueLocation> locations) =>
+            locations.Select(location =>
+            {
+                var localFilePath = absoluteFilePathLocator.Locate(location.FilePath);
+
+                if (location.TextRange == null)
+                {
+                    throw new ArgumentNullException(nameof(location.TextRange));
+                }
+
+                return new AnalysisIssueLocation(location.Message,
+                    localFilePath,
+                    location.TextRange.StartLine,
+                    location.TextRange.EndLine,
+                    location.TextRange.StartOffset,
+                    location.TextRange.EndOffset,
+                    null);
+            }).ToArray();
+
+        /// <summary>
+        /// Converts from the sonarqube issue severity enum to the standard AnalysisIssueSeverity
+        /// </summary>
+        internal /* for testing */ static AnalysisIssueSeverity Convert(SonarQubeIssueSeverity issueSeverity)
```

Comment:
Identical to https://github.com/SonarSource/sonarlint-visualstudio/blob/master/src/Integration.Vsix/SonarLintDaemon/SonarLintDaemon.cs#L389

Reply:
I don't think it's worth introducing a separate converter for this. However, the duplication isn't good either - a shared static class+method would be better.

Reply:
I'm sorry I should've clarified. It's identical in terms of logic, it's different enums. SonarQubeIssueSeverity vs DaemonIssueType.

Reply:
And the 'DaemonIssueType' is part of the gRPC message format so we can't share a single enum. Ok.